### PR TITLE
React cleanup

### PR DESCRIPTION
**Pull request by Bjvanminnen**
This cleans up some of our React components in code-studio, making it so that we treat them all fairly consistently. Some things I've done

(1)

```
var Foo = React.createClass({
...
module.exports = Foo;
```

instead of 
`var Foo = module.exports = React.createClass({`

The reasoning for this is to make it so that React tools (like the debugger tools in Chrome) think that the component is called `Foo` instead of `exports`. I had originally told Elijah that the latter approach was sufficient to accomplish this, but I was wrong (FYI @Hamms ).

(2)
Remove IFFEs. We no longer need these since we get them for free as part of the browserify step in code-studio

(3)
In cases where we insert our components into the dashboard namespace, do so at the end of the file. Mostly I just wanted to be consistent, and it felt natural to do it next to our module.exports assignment.

(4)
In the case of SmallFooter, expose the component itself on the dashboard namespace rather than exposing a render method. This was done mostly just so that we are consistent across all components.

---

In general, we still stick these onto the dashboard namespace more than I would like. I'm not entirely sure what the better long term solution is in all cases.

For cases where we're sharing with apps, it's probably to get these into a single pipeline, where apps can just require the component itself.

For cases where we're sharing with a dashboard js file, it's probably to get that js into code-studio.

For cases where we're sharing with a dashboard haml file, the situation is a little less clear. We could create a small js file that lives in code-studio, produces its own output file, and then have a script tag for that in haml. This is probably a cleaner approach, but seems like overkill in some cases.

### CODE REVIEW

File: code-studio/src/js/components/GridEditor.jsx [hunk 1]
```diff
@@ -141,3 +139,9 @@ window.dashboard.GridEditor = module.exports = React.createClass({
     </div>);
   },
 });
+module.exports = GridEditor;
+
+// We put this on the dashboard namespace because GridEditor is used by _grid.html.haml
+// TODO _ exposed just a renderer?
```
Comment (Bjvanminnen) [inline code-studio/src/js/components/GridEditor.jsx]: Brent, get rid of this TODO

File: code-studio/src/js/components/GridEditor.jsx [hunk 2]
```diff
@@ -141,3 +139,9 @@ window.dashboard.GridEditor = module.exports = React.createClass({
...
+window.dashboard = window.dashboard || {};
+window.dashboard.GridEditor = GridEditor;
```
Comment (Bjvanminnen) [inline code-studio/src/js/components/GridEditor.jsx:145]: For GridEditor, we could make this part of the levelbuilder.js output (which seems more correct), but then we will still need to stick this on a global if we want to access it from _grid.html.haml

**Comment by Bjvanminnen** [comment]
Things I tested:
Share page for applab and for a legacy level (star wars)
Share dialog when sharing from p/applab and from star wars
Footer on applab and starwars for share page an non-share page
SendToPhone in applab share dialog, and on share page for applab/starwars
Report abuse page
s/course1 (for progress components)

File: code-studio/src/js/components/GridEditor.jsx (hunk 2 above)
Comment (islemaster) [inline code-studio/src/js/components/GridEditor.jsx:145]: I've been tempted to have a separate `levelbuilder` global for things in the levelbuilder bundle that we need to access this way.

**Comment by islemaster** [comment]
Cool.  I don't have any concerns here.  It looks like a step forward, even with the remaining questions you call out. Thanks for cleaning these up! LGTM.

**Comment by Bjvanminnen** [comment]
Any thoughts on better ways to share with haml other than sticking on the global namespace?

**Comment by islemaster** [comment]
Don't we (supposedly) want to remove all JS from our views?  I think it's a little better to have a 'main'-esque file for a particular view that can attach needed handlers, etc, on document load.

For an example of this, see [levelbuilder_dsl.js](https://github.com/code-dot-org/code-dot-org/blob/staging/code-studio/src/js/levelbuilder_dsl.js), which gets [broken out as its own build product](https://github.com/code-dot-org/code-dot-org/blob/staging/code-studio/scripts/build-js.js#L34) and [included in the DSL editor page](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/views/levels/editors/_dsl.html.haml#L2).

**Pull request merged**